OpenQA/Isotovideo/NeedleDownloader.pm

```diff
@@ -0,0 +1,105 @@
+# Copyright © 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Isotovideo::NeedleDownloader;
+
+use strict;
+use warnings;
+use Mojo::Base -base;
+use Mojo::UserAgent;
+use Mojo::URL;
+use Mojo::File;
+use File::stat;
+use Try::Tiny;
+use POSIX 'strftime';
+use testapi 'diag';
+use bmwqemu;
+
+has files_to_download => sub { [] };
+has needle_dir        => sub { needle::default_needle_dir() };
+has openqa_url        => sub { 'http://' . $bmwqemu::vars{OPENQA_URL} };
+has ua                => sub { Mojo::UserAgent->new };
+
+sub _add_download {
+    my ($self, $needle, $extension, $path_param) = @_;
+
+    my $needle_name     = $needle->{name};
+    my $latest_update   = $needle->{t_updated};
+    my $needle_dir      = $self->needle_dir;
+    my $download_target = "$needle_dir/$needle_name.$extension";
+
+    if (my $target_stat = stat($download_target)) {
+        if (my $target_last_modified = $target_stat->[9] // $target_stat->[8]) {
+            $target_last_modified = strftime('%Y-%m-%dT%H:%M:%SZ', gmtime($target_last_modified));
+            if ($target_last_modified ge $latest_update) {
+                diag("skip downloading new needle: $download_target seems already up-to-date (last update: $target_last_modified > $latest_update)");
```

Comment:
I see imperative tense in logging messages as misleading. -> s/skip/Skipping/

Reply:
Right, I'll change that.